python/torch_mlir/fx.py

```diff
@@ -16,18 +16,58 @@
 from torch_mlir import ir
 from torch_mlir.dialects import torch as torch_d
 from torch_mlir.extras.fx_decomp_util import get_decomposition_table
+from torch_mlir.compiler_utils import (
+    OutputType,
+    run_pipeline_with_repro_report,
+    lower_mlir_module,
+)
+
+
+def _module_lowering(
+    verbose,
+    output_type,
+    torch_mod,
+    backend_legal_ops=None,
+    extra_library_file_name=None,
+):
+
+    if output_type == OutputType.TORCH:
+        if verbose:
+            print(torch_mod)
+        return torch_mod
+    # TODO: pass backend_legal_ops/extra_library_file_name by caller
+    if backend_legal_ops is None:
+        backend_legal_ops = []
+    if extra_library_file_name is None:
+        extra_library_file_name = ""
+    option_string = (
+        "{backend-legal-ops="
+        + ",".join(backend_legal_ops)
+        + " extra-library="
+        + extra_library_file_name
+        + "}"
+    )
+    run_pipeline_with_repro_report(
+        torch_mod,
+        f"builtin.module(torch-function-to-torch-backend-pipeline{option_string})",
+        "Lowering TorchFX IR -> Torch Backend IR",
+        enable_ir_printing=verbose,
+    )
```

Comment:
To digress a bit. Although the torch dialect module, which is directly from Dynamo/FX, theoretically does not require steps like decomposition and shape inference, there might be a situation where we decompose an operator into one that needs shape inference in the DecomposeComplexOpsPass, and this decomposed operator is not in dynamo's decomposition list. So, we can't temporarily strip off these passes, such as TorchShapeRefinement, in order to pass the corresponding case.
@stellaraccident

Reply:
I'm fine adding this stuff for now. We can evaluate/change course once we have better test coverage. Hard to do anything without that.